From 2c4a6215836fc2fda2abb28e115a8ba9d66d7859 Mon Sep 17 00:00:00 2001
From: Jeroen van Leerdam <vanLeerdam.Jeroen@kpmg.nl>
Date: Mon, 13 Mar 2017 11:09:05 +0100
Subject: [PATCH 1/7] CMake: Avoid linking to libs from Python install

The path of the Python installation found by the SearchInstalledSoftware
CMake module was automatically appended to the "CMAKE_PREFIX_PATH"
variable.  As a result, subsequent searches for external libraries
included this path and libraries from the Python installation are picked
up.  When using Python from Anaconda, this involves libraries such as
"readline", "crypto", and "ssl".

Linking to Anaconda libraries forces users to append the Anaconda
directory to the search path for shared libraries, which makes system
programs such as "wget", "curl", "ssh", and "gpg" load Anaconda
libraries as well.  This gives problems if the system libraries have the
same versions as the Anaconda libraries, but are slightly different.

The Python installation directory is not automatically appended to the
CMAKE_PREFIX_PATH any more.  If libraries or headers from this
installation are required for ROOT, the paths of these components can be
specified explicitly when configuring the ROOT installation.
---
 cmake/modules/SearchInstalledSoftware.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/modules/SearchInstalledSoftware.cmake b/cmake/modules/SearchInstalledSoftware.cmake
index eb8fc11..81c7dae 100644
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -422,7 +422,6 @@ if(python OR python3)
     message(STATUS "Found Python interpreter version ${PYTHON_VERSION}")
     execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import sys;sys.stdout.write(sys.prefix)"
                     OUTPUT_VARIABLE PYTHON_PREFIX)
-    set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} ${PYTHON_PREFIX})
   endif()
   set(Python_ADDITIONAL_VERSIONS ${PYTHON_VERSION})
   find_package(PythonLibs)
-- 
2.9.3

